src/app/components/SearchForm/WithSearch.jsx

```diff
@@ -25,6 +27,15 @@ function withSearch(WrappedComponent) {
 
     componentDidMount() {
       global.window.scrollTo(0, 0);
+      this.setState({ searchQuery: this.props.initialQuery });
+    }
+
+    componentDidUpdate(prevProps) {
+      // Wait for initial query to be returned before setting it in state
+      if (this.props.initialQuery && this.props.initialQuery !== prevProps.initialQuery) {
```

Comment:
This seems dumb and unnecessary, but it's not working without it.  Tagging @EdwinGuzman for a second opinion.